### DISCUSSION TITLE
Plot grains with arbitrary property

### DISCUSSION THREAD

**Discussion by nicholasbyres** [mtex-toolbox/mtex#853] [Ask Anything] [answered]
Dear community,

I have a list of grains (large_Grains) which are a subset of the original list of original grains. I have computed the "normalised KAM" value (sum of KAM of grain/grain size) for each of those grains and created a mx2 matrix that consists of the KAM value and the grainID of the grain that it corresponds to. I would now like to plot the grains on a map coloured by the newly computed "normalised KAM" value. 

Is it possible to plot the grains in a map, coloured by this "arbitrary" property?

Many thanks,
Nick

**Comment by zmichels** [comment]
Hi Nick,

This is definitely doable. For any set of grains, you can color them by a set of values of the same size/length. The order of the vector of values should match the order of the grains.

In a generic form, you want to use a syntax like the following:

plot(grains, value)

It sounds like you have everything you need. For example, if we refer to the different parts of your mx2 matrix as ’normKAM’ and ‘grainID’, you could try the following. 

plot(large_Grains(grainID),normKAM)

Cheers,
Zach

**Reply by nicholasbyres** [reply]
Hi Zach,

Great, thanks for your help. It worked. But I am a little confused. The syntax I used was:

plot(large_Grains,grain_KAM(:,1))

large_Grains being of grain2d and grain_KAM(:,1) being the list of normalised Kam values - so it turns out I don't need to use the grain ids at all? Is this because the list of grains are in the same order as the correspodning computed Kam values in the column vector? 

Thanks again for your help.

Kind regards,
Nick

**Comment by zmichels** [comment, marked as answer]
Hi Nick,

Yes, that’s exactly correct. But I wasn’t sure if that was the case or not (that the matrix and grains were still in same order), and so using the grainID’s would ensure they air paired correctly for plotting (based on what you mentioned about your mx2 matrix.

Also, something I have been doing more lately is appending the grain sets directly, so that any arbitrary things I compute are always directly associated with the grain from which I compute them. What I mean is… you can add any arbitrary property to a grain as follows.

% add the property and populate it
grains.prop.gKAM = grain_KAM;

% now use indexing to access it for something (in your case at the moment for plotting)
plot(grains, grains.gKAM)

% Now if you filter the grains at all, they keep that gKAM value in the subsequent/subset grainset
% example:
gSub = grains(grains.GOS>2*degree);
plot(gSub, gSmall.gKAM)

Cheers,
Zach

**Reply by nicholasbyres** [reply]
Amazing Zach! That is exactly what I've been looking for 😄 

Thanks again!
Nick

**Comment by zmichels** [comment]
oooops… last line typo.

should be:
plot(gSub, gSub.gKAM)
